failed to load `duckdb.lib` on Windows

### What happens?
Failed to load input file duckdb.lib on Windows while building on MSVC toolchains.

I reviewed past issues and found that most suggest installing MS VS Build Tools, but in fact, this did not solve my issue.

Im sure that I have installed MSVC and its toolchain, and configured them in the environment variables.

What also confused me is that on macOS, using C++ directly can compile DuckDB, but compiling with Rust will result in an error.

Below is the env path.
```
C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.44.35207\bin\Hostx64\x64
```
I added both to the environment variables and built the installation via `cargo add duckdb --features bundled`, but it still failed to load the input file "duckdb.lib". The error message is as follows.

### Error msg
```
error: linking with `link.exe` failed: exit code: 1181
  |
  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.44.35207\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "C:\\Users\\grayd\\AppData\\Local\\Temp\\rustcgIfEGZ\\symbols.o" "<78 object files omitted>" "C:\\Users\\grayd\\code\\fyke\\target\\debug\\deps/{libtonic_build-08b9567393e4c95b.rlib,libprost_build-0b8e5fc3f77fecca.rlib,libprettyplease-0703a93cf91f74ee.rlib,libsyn-859eb5bf8b13cbf9.rlib,libtempfile-335f2952791df61f.rlib,libfastrand-b0049bbb4869c479.rlib,libpetgraph-d9f85c758268ca0a.rlib,libfixedbitset-60e69d2dd442e055.rlib,libheck-d12e8df05c976497.rlib,libmultimap-8c333b38a8de3a4a.rlib,libitertools-e7687f9f62145f53.rlib,libeither-2d8e944c0515e859.rlib,libprost_types-b14ffd01a60ff82d.rlib,libquote-1485a27021ed9ce9.rlib,libproc_macro2-a46c307e964d513c.rlib,libunicode_ident-f935ccddac39a9e7.rlib}.rlib" "<sysroot>\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib/{libproc_macro-*,librustc_literal_escaper-*,librustc_std_workspace_std-*}.rlib" "C:\\Users\\grayd\\code\\fyke\\target\\debug\\deps/{libfyke_core-dc063920a29608c9.rlib,libduckdb-a26530c259b9f82e.rlib,libcast-49db356ef74b85f3.rlib,librust_decimal-3ba921d191be7d0d.rlib,libarrayvec-d50e6e157ab9318a.rlib,libfallible_streaming_iterator-04f60b605fc0f73f.rlib,libfallible_iterator-68563fc10ffcf3b8.rlib,libstrum-b2118dee6ffab062.rlib,libhashlink-3ba134715094e3cb.rlib,libarrow-e1b9a22b345bff6a.rlib,libarrow_row-91d05b69b70cbc66.rlib,libarrow_string-acff8fa3e3dfcd5b.rlib,libregex-710eef4d38ee3fd6.rlib,libregex_automata-db83f4512e1b5c7d.rlib,libaho_corasick-24adfb8458d9ade7.rlib,libregex_syntax-b6714b60b7bc7e4a.rlib,libmemchr-da6ba839bbe8ebcc.rlib,libarrow_cast-74a0889a8ddf2aa1.rlib,libatoi-e6ebe25009fa1943.rlib,libryu-37203d2c467f8da8.rlib,libcomfy_table-251418f61f10dac3.rlib,libunicode_segmentation-0b8b78906ba6799f.rlib,libunicode_width-5ad620de9423ed41.rlib,liblexical_core-a899d107d4a1caf3.rlib,liblexical_write_float-41eba6196db745d8.rlib,liblexical_write_integer-75f049f9e63a6664.rlib,liblexical_parse_float-43b8fed49b79bfbd.rlib,liblexical_parse_integer-8bdd1cf08a8114d2.rlib,liblexical_util-baf5b3dfc43aa25a.rlib,libstatic_assertions-da84d677a2cbb685.rlib,libarrow_arith-ea5b787e708c41ab.rlib,libarrow_ord-489e00c7ef23e751.rlib,libarrow_select-24e382ee34d058ed.rlib,libarrow_array-86d4625cfaaec43a.rlib,libahash-96dd9180461f4bfe.rlib,libgetrandom-1fba4a34be66d2cb.rlib,libzerocopy-e1dffdfb07107702.rlib,libchrono-1b2101bbf07384ce.rlib,libwindows_link-71c4caf817584dd3.rlib,libarrow_data-a31197f5464dcbdc.rlib,libarrow_schema-f4ee27f94a7e2fd1.rlib,libbitflags-05ed3484cc55db94.rlib,libarrow_buffer-a0c15988dc061bdc.rlib,libhalf-747f7d890291de0f.rlib,libcfg_if-3d39ba26ec35ebac.rlib,libnum-f0aa498cc9e9b9e3.rlib,libnum_iter-b389ef6f565c05d2.rlib,libnum_rational-1b7cd3981135e2f8.rlib,libnum_complex-2537c84c16f51dd3.rlib,libnum_bigint-737a93fc7600053a.rlib,libnum_integer-e7cba3faece1f759.rlib,libnum_traits-1701bc9abe802865.rlib,liblibduckdb_sys-3ee4c5fa7b659cec.rlib,libtonic-63ccc808f358713e.rlib,librustls_native_certs-9521c9c9a12fcb54.rlib,libschannel-b3dae31ae879a1d9.rlib,libhyper_timeout-d4b3544e13b4ed67.rlib,libtokio_rustls-2882d3693e439f06.rlib,librustls-33273b356406a39e.rlib,libsubtle-c12f9f250859eba5.rlib,libwebpki-722a7fe3cf271009.rlib,libuntrusted-101ebd898726120a.rlib,librustls_pki_types-60acb31f7a26a7c7.rlib,libzeroize-681184fc434f5fa1.rlib,liblog-951db6d18ef7a2b2.rlib,libhyper_util-cb9ae073e580524c.rlib,libhyper-bb87749ee6aa1a3e.rlib,libwant-8806602f42f40509.rlib,libtry_lock-34576e8b240dd8ad.rlib,libsmallvec-058dce38b90e64ce.rlib,libhttparse-7eebae40f3537ccd.rlib,libh2-3e49d7846549ee60.rlib,libslab-c7f7dfa5f20b7f04.rlib,libatomic_waker-733411725d00cdd1.rlib,libhttpdate-5729e4e4bfc91a57.rlib,libfutures_channel-e3d3683f3718bdf5.rlib,libaxum-e95e719de6ac7194.rlib,libpercent_encoding-7b1b5cbfc31ea8e2.rlib,libmatchit-6a235b64c4b6e181.rlib,libserde-bfd67cd6c68e360a.rlib,libaxum_core-cc75f97c92501569.rlib,libmime-7ad6c1913bc944e2.rlib,libtower-7b5f2c57ea426ef4.rlib,libsync_wrapper-c9414faca8baf90a.rlib,libindexmap-fc1fe7a29aa9dadb.rlib,libequivalent-43bf456d570bccb8.rlib,libhashbrown-5a1847c4e38fdd05.rlib,libfoldhash-c7e1bc489b06a6d3.rlib,libtokio_util-375118be2f1e8f12.rlib,libfutures_sink-a6d190d3a8fcb8b8.rlib,libfutures_util-86853a2efbd1a964.rlib,libfutures_task-a81ede2821d2f383.rlib,libpin_utils-cdf15da9de6652f6.rlib,libtower_layer-562bf17403e4f363.rlib,libbase64-0b360acb9dc29ba9.rlib,libprost-59338344ba55e47a.rlib,libpin_project-c81cfdfd99e0ed3c.rlib,libtracing-2f3fd1894abcf2c1.rlib,libtracing_core-cce52fed0875b35e.rlib,libonce_cell-47fad646ee02dfb9.rlib,libzstd-f9972ab341b7ddcc.rlib,libzstd_safe-903d725809f14243.rlib,libzstd_sys-e7846f37eb31065b.rlib,libtower_service-c25440accab56b57.rlib,libtokio_stream-d181710f07ee752a.rlib,libtokio-2c299133e13c049e.rlib,libsocket2-2b41d1da27ca4b26.rlib,libwindows_sys-508a27a03b7e04c1.rlib,libmio-08874e0113789897.rlib,libwindows_sys-276407dfe879ff5f.rlib,libwindows_targets-e7bdccf469b35d45.rlib,libhttp_body_util-c1a0329cd73bb4c4.rlib,libpin_project_lite-c2856b841f2f0624.rlib,libfutures_core-1081b66f01af0139.rlib,libhttp_body-1241c5258b3dc20b.rlib,libhttp-fe373aff1fe7a821.rlib,libitoa-6368bc7d7734ae28.rlib,libfnv-a847c0352ba4ddbf.rlib,libbytes-441da4d76b13cbcb.rlib,libthiserror-e7d7309ca70bb9bb.rlib}.rlib" "<sysroot>\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib/{libstd-*,libpanic_unwind-*,libwindows_targets-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libunwind-*,libcfg_if-*,liballoc-*,librustc_std_workspace_core-*,libcore-*,libcompiler_builtins-*}.rlib" "duckdb.lib" "D:\\rust\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib\\windows.0.52.0.lib" "D:\\rust\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib\\windows.0.52.0.lib" "kernel32.lib" "kernel32.lib" "ntdll.lib" "userenv.lib" "ws2_32.lib" "dbghelp.lib" "/defaultlib:msvcrt" "/NXCOMPAT" "/LIBPATH:C:\\Users\\grayd\\code\\fyke\\target\\debug\\build\\zstd-sys-eb06c31bd2391b0e\\out" "/LIBPATH:D:\\rust\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib" "/LIBPATH:C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.44.35207\\atlmfc\\lib\\x64" "/OUT:C:\\Users\\grayd\\code\\fyke\\target\\debug\\build\\fyke-grpc-866672fa1570fc82\\build_script_build-866672fa1570fc82.exe" "/OPT:REF,NOICF" "/DEBUG" "/PDBALTPATH:%_PDB%" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\libstd.natvis"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: LINK : fatal error LNK1181: cannot open input file “duckdb.lib”␍
```
### Pretty Error msg Print
```
error: linking with `link.exe` failed: exit code: 1181
  |
  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.44.35207\\bin\\HostX64\\x64\\link.exe" 
          "/NOLOGO" 
          "C:\\Users\\grayd\\AppData\\Local\\Temp\\rustcgIfEGZ\\symbols.o" 
          "<78 object files omitted>" 
          "C:\\Users\\grayd\\code\\fyke\\target\\debug\\deps/{libtonic_build-08b9567393e4c95b.rlib,libprost_build-0b8e5fc3f77fecca.rlib,libprettyplease-0703a93cf91f74ee.rlib,libsyn-859eb5bf8b13cbf9.rlib,libtempfile-335f2952791df61f.rlib,libfastrand-b0049bbb4869c479.rlib,libpetgraph-d9f85c758268ca0a.rlib,libfixedbitset-60e69d2dd442e055.rlib,libheck-d12e8df05c976497.rlib,libmultimap-8c333b38a8de3a4a.rlib,libitertools-e7687f9f62145f53.rlib,libeither-2d8e944c0515e859.rlib,libprost_types-b14ffd01a60ff82d.rlib,libquote-1485a27021ed9ce9.rlib,libproc_macro2-a46c307e964d513c.rlib,libunicode_ident-f935ccddac39a9e7.rlib}.rlib" 
          "<sysroot>\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib/{libproc_macro-*,librustc_literal_escaper-*,librustc_std_workspace_std-*}.rlib" 
          "C:\\Users\\grayd\\code\\fyke\\target\\debug\\deps/{libfyke_core-dc063920a29608c9.rlib,libduckdb-a26530c259b9f82e.rlib,libcast-49db356ef74b85f3.rlib,librust_decimal-3ba921d191be7d0d.rlib,libarrayvec-d50e6e157ab9318a.rlib,libfallible_streaming_iterator-04f60b605fc0f73f.rlib,libfallible_iterator-68563fc10ffcf3b8.rlib,libstrum-b2118dee6ffab062.rlib,libhashlink-3ba134715094e3cb.rlib,libarrow-e1b9a22b345bff6a.rlib,libarrow_row-91d05b69b70cbc66.rlib,libarrow_string-acff8fa3e3dfcd5b.rlib,libregex-710eef4d38ee3fd6.rlib,libregex_automata-db83f4512e1b5c7d.rlib,libaho_corasick-24adfb8458d9ade7.rlib,libregex_syntax-b6714b60b7bc7e4a.rlib,libmemchr-da6ba839bbe8ebcc.rlib,libarrow_cast-74a0889a8ddf2aa1.rlib,libatoi-e6ebe25009fa1943.rlib,libryu-37203d2c467f8da8.rlib,libcomfy_table-251418f61f10dac3.rlib,libunicode_segmentation-0b8b78906ba6799f.rlib,libunicode_width-5ad620de9423ed41.rlib,liblexical_core-a899d107d4a1caf3.rlib,liblexical_write_float-41eba6196db745d8.rlib,liblexical_write_integer-75f049f9e63a6664.rlib,liblexical_parse_float-43b8fed49b79bfbd.rlib,liblexical_parse_integer-8bdd1cf08a8114d2.rlib,liblexical_util-baf5b3dfc43aa25a.rlib,libstatic_assertions-da84d677a2cbb685.rlib,libarrow_arith-ea5b787e708c41ab.rlib,libarrow_ord-489e00c7ef23e751.rlib,libarrow_select-24e382ee34d058ed.rlib,libarrow_array-86d4625cfaaec43a.rlib,libahash-96dd9180461f4bfe.rlib,libgetrandom-1fba4a34be66d2cb.rlib,libzerocopy-e1dffdfb07107702.rlib,libchrono-1b2101bbf07384ce.rlib,libwindows_link-71c4caf817584dd3.rlib,libarrow_data-a31197f5464dcbdc.rlib,libarrow_schema-f4ee27f94a7e2fd1.rlib,libbitflags-05ed3484cc55db94.rlib,libarrow_buffer-a0c15988dc061bdc.rlib,libhalf-747f7d890291de0f.rlib,libcfg_if-3d39ba26ec35ebac.rlib,libnum-f0aa498cc9e9b9e3.rlib,libnum_iter-b389ef6f565c05d2.rlib,libnum_rational-1b7cd3981135e2f8.rlib,libnum_complex-2537c84c16f51dd3.rlib,libnum_bigint-737a93fc7600053a.rlib,libnum_integer-e7cba3faece1f759.rlib,libnum_traits-1701bc9abe802865.rlib,liblibduckdb_sys-3ee4c5fa7b659cec.rlib,libtonic-63ccc808f358713e.rlib,librustls_native_certs-9521c9c9a12fcb54.rlib,libschannel-b3dae31ae879a1d9.rlib,libhyper_timeout-d4b3544e13b4ed67.rlib,libtokio_rustls-2882d3693e439f06.rlib,librustls-33273b356406a39e.rlib,libsubtle-c12f9f250859eba5.rlib,libwebpki-722a7fe3cf271009.rlib,libuntrusted-101ebd898726120a.rlib,librustls_pki_types-60acb31f7a26a7c7.rlib,libzeroize-681184fc434f5fa1.rlib,liblog-951db6d18ef7a2b2.rlib,libhyper_util-cb9ae073e580524c.rlib,libhyper-bb87749ee6aa1a3e.rlib,libwant-8806602f42f40509.rlib,libtry_lock-34576e8b240dd8ad.rlib,libsmallvec-058dce38b90e64ce.rlib,libhttparse-7eebae40f3537ccd.rlib,libh2-3e49d7846549ee60.rlib,libslab-c7f7dfa5f20b7f04.rlib,libatomic_waker-733411725d00cdd1.rlib,libhttpdate-5729e4e4bfc91a57.rlib,libfutures_channel-e3d3683f3718bdf5.rlib,libaxum-e95e719de6ac7194.rlib,libpercent_encoding-7b1b5cbfc31ea8e2.rlib,libmatchit-6a235b64c4b6e181.rlib,libserde-bfd67cd6c68e360a.rlib,libaxum_core-cc75f97c92501569.rlib,libmime-7ad6c1913bc944e2.rlib,libtower-7b5f2c57ea426ef4.rlib,libsync_wrapper-c9414faca8baf90a.rlib,libindexmap-fc1fe7a29aa9dadb.rlib,libequivalent-43bf456d570bccb8.rlib,libhashbrown-5a1847c4e38fdd05.rlib,libfoldhash-c7e1bc489b06a6d3.rlib,libtokio_util-375118be2f1e8f12.rlib,libfutures_sink-a6d190d3a8fcb8b8.rlib,libfutures_util-86853a2efbd1a964.rlib,libfutures_task-a81ede2821d2f383.rlib,libpin_utils-cdf15da9de6652f6.rlib,libtower_layer-562bf17403e4f363.rlib,libbase64-0b360acb9dc29ba9.rlib,libprost-59338344ba55e47a.rlib,libpin_project-c81cfdfd99e0ed3c.rlib,libtracing-2f3fd1894abcf2c1.rlib,libtracing_core-cce52fed0875b35e.rlib,libonce_cell-47fad646ee02dfb9.rlib,libzstd-f9972ab341b7ddcc.rlib,libzstd_safe-903d725809f14243.rlib,libzstd_sys-e7846f37eb31065b.rlib,libtower_service-c25440accab56b57.rlib,libtokio_stream-d181710f07ee752a.rlib,libtokio-2c299133e13c049e.rlib,libsocket2-2b41d1da27ca4b26.rlib,libwindows_sys-508a27a03b7e04c1.rlib,libmio-08874e0113789897.rlib,libwindows_sys-276407dfe879ff5f.rlib,libwindows_targets-e7bdccf469b35d45.rlib,libhttp_body_util-c1a0329cd73bb4c4.rlib,libpin_project_lite-c2856b841f2f0624.rlib,libfutures_core-1081b66f01af0139.rlib,libhttp_body-1241c5258b3dc20b.rlib,libhttp-fe373aff1fe7a821.rlib,libitoa-6368bc7d7734ae28.rlib,libfnv-a847c0352ba4ddbf.rlib,libbytes-441da4d76b13cbcb.rlib,libthiserror-e7d7309ca70bb9bb.rlib}.rlib" 
          "<sysroot>\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib/{libstd-*,libpanic_unwind-*,libwindows_targets-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libunwind-*,libcfg_if-*,liballoc-*,librustc_std_workspace_core-*,libcore-*,libcompiler_builtins-*}.rlib" 
          "duckdb.lib" 
          "D:\\rust\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib\\windows.0.52.0.lib" 
          "D:\\rust\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib\\windows.0.52.0.lib" 
          "kernel32.lib" 
          "kernel32.lib" 
          "ntdll.lib" 
          "userenv.lib" 
          "ws2_32.lib" 
          "dbghelp.lib" 
          "/defaultlib:msvcrt" 
          "/NXCOMPAT" 
          "/LIBPATH:C:\\Users\\grayd\\code\\fyke\\target\\debug\\build\\zstd-sys-eb06c31bd2391b0e\\out" 
          "/LIBPATH:D:\\rust\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib" 
          "/LIBPATH:C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.44.35207\\atlmfc\\lib\\x64" 
          "/OUT:C:\\Users\\grayd\\code\\fyke\\target\\debug\\build\\fyke-grpc-866672fa1570fc82\\build_script_build-866672fa1570fc82.exe" 
          "/OPT:REF,NOICF" 
          "/DEBUG" 
          "/PDBALTPATH:%_PDB%" 
          "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\intrinsic.natvis" 
          "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\liballoc.natvis" 
          "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\libcore.natvis" 
          "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\libstd.natvis"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: LINK : fatal error LNK1181: cannot open input file “duckdb.lib”
```



### To Reproduce

This issue is consistently reproducible on my Windows system. First, I installed `MS Visual Studio C++ Build Tools` and added the paths for CMake, Cl, etc., to the environment variables. Then, I installed the `MSVC toolchain for Rust 1.88.0` and created a project. I added duckdb using the command cargo add duckdb --features bundled, but unfortunately, I always encounter an error during cargo build.

Perhaps I missed some critical steps, but the above is the complete process to reproduce the issue on my Windows system.

### Below is the code
```
use chrono::prelude::*;
use duckdb::{Appender, Connection};
use fyke_core::Result;

#[derive(Debug)]
pub struct UserStat {
    pub id: i64,
    pub email: String,
    pub name: String,
    pub created_at: DateTime<Utc>,
    pub last_visited_at: Option<DateTime<Utc>>,
    pub last_watched_at: Option<DateTime<Utc>>,
    pub recent_watched: Option<Vec<i64>>,
    pub recorded_not_viewed: Option<Vec<i64>>,
    pub viewed_not_done: Option<Vec<i64>>,
    pub done: Option<Vec<i64>>,
}

impl UserStat {
    pub fn connect() -> Result<()> {
        let conn = Connection::open_in_memory()?;
        conn.execute_batch(
            r"
            CREATE SEQUENCE seq;
            CREATE TABLE user_stat (
                id BIGINT PRIMARY KEY DEFAULT NEXTVAL('seq'),
                email VARCHAR(128) NOT NULL,
                name VARCHAR(64) NOT NULL,
                created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
                last_visited_at TIMESTAMPTZ,
                last_watched_at TIMESTAMPTZ,
                recent_watched BIGINT[],
                recorded_not_viewed BIGINT[],
                viewed_not_done BIGINT[],
                done BIGINT[]
            );
            ",
        );

        Ok(())
    }
}
```

### Rustc ver
```
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: x86_64-pc-windows-msvc
release: 1.88.0
LLVM version: 20.1.5
```

### OS:

Windows11 24h2

### DuckDB Version:

1.3.2

### DuckDB Client:

Rust

--- a/crates/libduckdb-sys/build.rs
+++ b/crates/libduckdb-sys/build.rs
@@ -56,12 +56,10 @@ pub(crate) fn write_bindings(header_dir: &Path, out_path: &Path) {
     }
 }
 
-/// Link the Windows system libraries DuckDB needs but that neither `cc` nor a static
-/// CMake link adds automatically. Mirrors DuckDB's `DUCKDB_SYSTEM_LIBS`
-/// (duckdb-sources/src/CMakeLists.txt): `ws2_32` + `rstrtmgr` under `if(MSVC OR MINGW)`
-/// and `bcrypt` under `if(MSVC)`; the MinGW path also needs `stdc++`. Single canonical
-/// copy shared by both bundled backends — keep in sync with that upstream list on
-/// submodule bumps. Callers apply their own `win_target()` gate.
+/// Link the Windows system libraries DuckDB needs but that neither `cc` nor a
+/// static CMake link adds automatically. Mirrors DuckDB's `DUCKDB_SYSTEM_LIBS`
+/// (duckdb-sources/src/CMakeLists.txt) — keep in sync on submodule bumps.
+/// Callers apply their own `win_target()` gate.
 #[cfg(feature = "bundled")]
 pub(crate) fn link_windows_system_libs() {
     println!("cargo:rustc-link-lib=dylib=ws2_32");

--- a/crates/libduckdb-sys/build.rs
+++ b/crates/libduckdb-sys/build.rs
@@ -17,7 +17,6 @@ fn win_target() -> bool {
 /// the content of `CARGO_CFG_TARGET_ENV` (and is always lowercase)
 ///
 /// See [`win_target`]
-#[cfg(any(not(feature = "bundled"), feature = "bundled-cmake"))]
 fn is_compiler(compiler_name: &str) -> bool {
     std::env::var("CARGO_CFG_TARGET_ENV").is_ok_and(|v| v == compiler_name)
 }
@@ -54,6 +53,23 @@ pub(crate) fn write_bindings(header_dir: &Path, out_path: &Path) {
     {
         let _ = header_dir;
         copy_pregenerated_bindings(out_path);
+    }
+}
+
+/// Link the Windows system libraries DuckDB needs but that neither `cc` nor a static
+/// CMake link adds automatically. Mirrors DuckDB's `DUCKDB_SYSTEM_LIBS`
+/// (duckdb-sources/src/CMakeLists.txt): `ws2_32` + `rstrtmgr` under `if(MSVC OR MINGW)`
+/// and `bcrypt` under `if(MSVC)`; the MinGW path also needs `stdc++`. Single canonical
+/// copy shared by both bundled backends — keep in sync with that upstream list on
+/// submodule bumps. Callers apply their own `win_target()` gate.
+#[cfg(feature = "bundled")]
+pub(crate) fn link_windows_system_libs() {
+    println!("cargo:rustc-link-lib=dylib=ws2_32");
+    println!("cargo:rustc-link-lib=dylib=rstrtmgr");
+    if is_compiler("msvc") {
+        println!("cargo:rustc-link-lib=dylib=bcrypt");
+    } else {
+        println!("cargo:rustc-link-lib=dylib=stdc++");
     }
 }
 

--- a/crates/libduckdb-sys/build_bundled_cc.rs
+++ b/crates/libduckdb-sys/build_bundled_cc.rs
@@ -1,4 +1,4 @@
-use crate::{link_windows_system_libs, win_target, write_bindings};
+use crate::{is_compiler, link_windows_system_libs, win_target, write_bindings};
 use std::{
     collections::{HashMap, HashSet},
     path::Path,
@@ -113,6 +113,20 @@ pub fn main(out_dir: &str, out_path: &Path) {
         .flag_if_supported("/bigobj")
         .warnings(false)
         .flag_if_supported("-w");
+
+    // Enable C++ exceptions on MSVC. CMake-built DuckDB gets /EHsc from CMake's default
+    // MSVC flags, but the cc build must add it explicitly: without it MSVC leaves
+    // exception handling off, the C API's try/catch blocks are inert, and a thrown
+    // exception (e.g. an invalid-UTF-8 CSV read) aborts the process with
+    // STATUS_STACK_BUFFER_OVERRUN (https://github.com/duckdb/duckdb-rs/issues/774).
+    // Apply it as a hard, gated flag rather than flag_if_supported: /EHsc is always
+    // accepted by MSVC/clang-cl, so the probe can only ever pass — but a *spurious*
+    // probe failure would silently drop /EHsc, yielding a clean build that crashes at
+    // runtime with no signal. gcc/clang (incl. MinGW) enable exceptions by default and
+    // reject /EHsc, so gate on MSVC. Mirrors the gate in build_bundled_cmake.rs.
+    if win_target() && is_compiler("msvc") {
+        cfg.flag("/EHsc");
+    }
 
     let is_debug = match std::env::var("DEBUG") {
         Ok(v) => v != "false" && v != "0",

--- a/crates/libduckdb-sys/build_bundled_cc.rs
+++ b/crates/libduckdb-sys/build_bundled_cc.rs
@@ -114,16 +114,12 @@ pub fn main(out_dir: &str, out_path: &Path) {
         .warnings(false)
         .flag_if_supported("-w");
 
-    // Enable C++ exceptions on MSVC. CMake-built DuckDB gets /EHsc from CMake's default
-    // MSVC flags, but the cc build must add it explicitly: without it MSVC leaves
-    // exception handling off, the C API's try/catch blocks are inert, and a thrown
-    // exception (e.g. an invalid-UTF-8 CSV read) aborts the process with
-    // STATUS_STACK_BUFFER_OVERRUN (https://github.com/duckdb/duckdb-rs/issues/774).
-    // Apply it as a hard, gated flag rather than flag_if_supported: /EHsc is always
-    // accepted by MSVC/clang-cl, so the probe can only ever pass — but a *spurious*
-    // probe failure would silently drop /EHsc, yielding a clean build that crashes at
-    // runtime with no signal. gcc/clang (incl. MinGW) enable exceptions by default and
-    // reject /EHsc, so gate on MSVC. Mirrors the gate in build_bundled_cmake.rs.
+    // Enable C++ exceptions on MSVC: without /EHsc, exception handling is off, the
+    // C API's try/catch blocks are inert, and a thrown exception (e.g. invalid-UTF-8
+    // CSV read) aborts with STATUS_STACK_BUFFER_OVERRUN (#774). Hard flag, not
+    // flag_if_supported, so a spurious probe miss can't silently drop it. gcc/clang
+    // enable exceptions by default and reject /EHsc, so gate on MSVC. Mirrors
+    // build_bundled_cmake.rs.
     if win_target() && is_compiler("msvc") {
         cfg.flag("/EHsc");
     }
@@ -141,9 +137,8 @@ pub fn main(out_dir: &str, out_path: &Path) {
     }
     cfg.compile("duckdb");
 
-    // DuckDB references Windows system libraries that `cc` does not link automatically
-    // (e.g. unresolved `RmStartSession` pulled in by `duckdb::AdditionalLockInfo`, the
-    // Restart Manager). See link_windows_system_libs for the list and upstream mapping.
+    // `cc` does not link DuckDB's Windows system libs automatically (e.g. unresolved
+    // `RmStartSession` from the Restart Manager). See link_windows_system_libs.
     if win_target() {
         link_windows_system_libs();
     }

--- a/crates/libduckdb-sys/build_bundled_cc.rs
+++ b/crates/libduckdb-sys/build_bundled_cc.rs
@@ -1,4 +1,4 @@
-use crate::{win_target, write_bindings};
+use crate::{link_windows_system_libs, win_target, write_bindings};
 use std::{
     collections::{HashMap, HashSet},
     path::Path,
@@ -126,6 +126,13 @@ pub fn main(out_dir: &str, out_path: &Path) {
         cfg.define("DUCKDB_BUILD_LIBRARY", None);
     }
     cfg.compile("duckdb");
+
+    // DuckDB references Windows system libraries that `cc` does not link automatically
+    // (e.g. unresolved `RmStartSession` pulled in by `duckdb::AdditionalLockInfo`, the
+    // Restart Manager). See link_windows_system_libs for the list and upstream mapping.
+    if win_target() {
+        link_windows_system_libs();
+    }
 
     println!("cargo:lib_dir={out_dir}");
 }

--- a/crates/libduckdb-sys/build_bundled_cmake.rs
+++ b/crates/libduckdb-sys/build_bundled_cmake.rs
@@ -1,4 +1,4 @@
-use crate::{is_compiler, win_target, write_bindings};
+use crate::{is_compiler, link_windows_system_libs, win_target, write_bindings};
 use std::{
     collections::BTreeSet,
     env::{self, VarError},
@@ -469,15 +469,7 @@ fn link_system_libs() {
         "macos" => {
             println!("cargo:rustc-link-lib=dylib=c++");
         }
-        "windows" => {
-            println!("cargo:rustc-link-lib=dylib=ws2_32");
-            println!("cargo:rustc-link-lib=dylib=rstrtmgr");
-            if is_compiler("msvc") {
-                println!("cargo:rustc-link-lib=dylib=bcrypt");
-            } else {
-                println!("cargo:rustc-link-lib=dylib=stdc++");
-            }
-        }
+        "windows" => link_windows_system_libs(),
         other => {
             panic!(
                 "bundled-cmake is currently supported only on Linux, macOS, and Windows; unsupported target OS `{other}`"

--- a/crates/libduckdb-sys/build_bundled_cmake.rs
+++ b/crates/libduckdb-sys/build_bundled_cmake.rs
@@ -134,6 +134,12 @@ pub fn main(out_dir: &str, out_path: &Path) {
             if disable_extension_load { "0" } else { "1" },
         );
 
+    // Windows MAX_PATH caveat: this build emits very deep object paths (the
+    // compressed_materialization unity object runs ~150 chars below the build dir) and
+    // Ninja does not shorten them to CMAKE_OBJECT_PATH_MAX, so a deep OUT_DIR can exceed
+    // 260 chars and fail late with `C1083: Cannot open compiler generated file: ''`. The
+    // build script can't raise the limit or move OUT_DIR — shorten the path instead
+    // (short CARGO_TARGET_DIR, drop `--target`, or check out nearer the drive root).
     let dst = config.build();
     let lib_dir = dst.join("lib");
     validate_extension_libraries(&lib_dir, &cmake_build_type, &enabled_extensions);

--- a/crates/libduckdb-sys/build_bundled_cmake.rs
+++ b/crates/libduckdb-sys/build_bundled_cmake.rs
@@ -70,22 +70,14 @@ pub fn main(out_dir: &str, out_path: &Path) {
         .define("CMAKE_C_FLAGS_INIT", warning_suppression_flag())
         .define("CMAKE_CXX_FLAGS_INIT", warning_suppression_flag());
 
-    // Re-enable C++ exceptions on MSVC. CMake's platform default would add /EHsc to
-    // CMAKE_CXX_FLAGS, but the cmake crate makes its own `cxxflag`s the *source* of
-    // -DCMAKE_CXX_FLAGS, replacing CMake's default, so /EHsc is otherwise dropped and
-    // DuckDB compiles with exception handling effectively off. A single-threaded error
-    // still unwinds via x64's table-based mechanism, but an exception thrown across the
-    // parallel CSV reader's worker threads (e.g. invalid UTF-8) is then mishandled at
-    // the C API boundary: it aborts with STATUS_STACK_BUFFER_OVERRUN (0xC0000409) under
-    // a CRT mismatch, or deadlocks when the CRT is aligned, instead of returning an
-    // error (https://github.com/duckdb/duckdb-rs/issues/774).
+    // Re-enable C++ exceptions on MSVC. The cmake crate makes its `cxxflag`s the
+    // source of -DCMAKE_CXX_FLAGS, replacing CMake's default (which includes /EHsc),
+    // so exception handling ends up effectively off: a thrown exception (e.g.
+    // invalid-UTF-8 CSV read) then aborts with STATUS_STACK_BUFFER_OVERRUN (#774).
     //
-    // INVARIANT: cxxflag("/EHsc") only reaches the compiler because we never call
-    // `config.define("CMAKE_CXX_FLAGS", ..)` — the cmake crate sources -DCMAKE_CXX_FLAGS
-    // from its `cxxflags` *only when* that var is not already in `defines`. Defining
-    // CMAKE_CXX_FLAGS anywhere here would silently discard /EHsc and reintroduce #774.
-    // (CMAKE_CXX_FLAGS_INIT, which we set to "/w", is a different variable and is safe.)
-    // Mirrors the gate in build_bundled_cc.rs; gcc/clang enable exceptions by default.
+    // Don't `config.define("CMAKE_CXX_FLAGS", ..)` here: it overrides the cxxflag
+    // above and silently drops /EHsc, reintroducing #774.
+    // Mirrors build_bundled_cc.rs; gcc/clang enable exceptions by default.
     if win_target() && is_compiler("msvc") {
         config.cxxflag("/EHsc");
     }
@@ -134,12 +126,11 @@ pub fn main(out_dir: &str, out_path: &Path) {
             if disable_extension_load { "0" } else { "1" },
         );
 
-    // Windows MAX_PATH caveat: this build emits very deep object paths (the
-    // compressed_materialization unity object runs ~150 chars below the build dir) and
-    // Ninja does not shorten them to CMAKE_OBJECT_PATH_MAX, so a deep OUT_DIR can exceed
-    // 260 chars and fail late with `C1083: Cannot open compiler generated file: ''`. The
-    // build script can't raise the limit or move OUT_DIR — shorten the path instead
-    // (short CARGO_TARGET_DIR, drop `--target`, or check out nearer the drive root).
+    // Windows MAX_PATH caveat: this build emits very deep object paths (Ninja ignores
+    // CMAKE_OBJECT_PATH_MAX), so a deep OUT_DIR can exceed 260 chars and fail late with
+    // `C1083: Cannot open compiler generated file: ''`. The build script can't move
+    // OUT_DIR — shorten the path (short CARGO_TARGET_DIR, drop `--target`, or check out
+    // nearer the drive root).
     let dst = config.build();
     let lib_dir = dst.join("lib");
     validate_extension_libraries(&lib_dir, &cmake_build_type, &enabled_extensions);

--- a/crates/libduckdb-sys/build_bundled_cmake.rs
+++ b/crates/libduckdb-sys/build_bundled_cmake.rs
@@ -70,6 +70,26 @@ pub fn main(out_dir: &str, out_path: &Path) {
         .define("CMAKE_C_FLAGS_INIT", warning_suppression_flag())
         .define("CMAKE_CXX_FLAGS_INIT", warning_suppression_flag());
 
+    // Re-enable C++ exceptions on MSVC. CMake's platform default would add /EHsc to
+    // CMAKE_CXX_FLAGS, but the cmake crate makes its own `cxxflag`s the *source* of
+    // -DCMAKE_CXX_FLAGS, replacing CMake's default, so /EHsc is otherwise dropped and
+    // DuckDB compiles with exception handling effectively off. A single-threaded error
+    // still unwinds via x64's table-based mechanism, but an exception thrown across the
+    // parallel CSV reader's worker threads (e.g. invalid UTF-8) is then mishandled at
+    // the C API boundary: it aborts with STATUS_STACK_BUFFER_OVERRUN (0xC0000409) under
+    // a CRT mismatch, or deadlocks when the CRT is aligned, instead of returning an
+    // error (https://github.com/duckdb/duckdb-rs/issues/774).
+    //
+    // INVARIANT: cxxflag("/EHsc") only reaches the compiler because we never call
+    // `config.define("CMAKE_CXX_FLAGS", ..)` — the cmake crate sources -DCMAKE_CXX_FLAGS
+    // from its `cxxflags` *only when* that var is not already in `defines`. Defining
+    // CMAKE_CXX_FLAGS anywhere here would silently discard /EHsc and reintroduce #774.
+    // (CMAKE_CXX_FLAGS_INIT, which we set to "/w", is a different variable and is safe.)
+    // Mirrors the gate in build_bundled_cc.rs; gcc/clang enable exceptions by default.
+    if win_target() && is_compiler("msvc") {
+        config.cxxflag("/EHsc");
+    }
+
     // Unity builds (DuckDB's default) combine .cpp files into fewer translation
     // units and compile significantly faster. Allow opting out for debugging.
     // Always set explicitly so the CMake cache doesn't keep a stale value.

